File: advent_of_code_2023/1-Trebuchet/part_1.py

def get_calibration_number(string):
    digits = []
    for char in string:
        if char.isdigit():
            digits.append(int(char))
    return 10 * digits[0] + digits[-1]

if __name__ == '__main__':
    with open("puzzle_input.txt", "r") as file:
        puzzle_input = file.read().split("\n")

    print(puzzle_input)

    print(sum(get_calibration_number(string) for string in puzzle_input))

File: advent_of_code_2023/4-Scratchards/part_1.js
const fs = require("fs")

// Read the file synchronously
const filePath = "./puzzle_input.txt" // path to your puzzle_input.txt
const fileContent = fs.readFileSync(filePath, "utf-8")

const scratchcards = fileContent.split("\n").map((line) =>
    line
        .slice(9)
        .split("|")
        .map((side) => side.trim().split(/\s+/).map((numberString) => parseInt(numberString)))
)

let total = 0

for (let [winningNumbers, myNumbers] of scratchcards) {

    let scratchcardTotal = 0

    for (let winningNumber of winningNumbers) {
        if (myNumbers.includes(winningNumber)) {
            scratchcardTotal += 1
        }
    }
    
    if (scratchcardTotal > 0) {
        total += 2 ** (scratchcardTotal - 1)
    }
}

console.log(total)